service/lib/dinstaller/dbus/users.rb

```diff
@@ -90,13 +90,20 @@ def initialize(backend, logger)
         end
 
         dbus_method :SetFirstUser,
-          FUSER_SIG + ", out result:u" do |full_name, user_name, password, auto_login, data|
+          # It returns an Struct with the first field with the result of the operation as a boolean
+          # and the second parameter as an array of issues found in case of failure
```

Comment:
Well, thanks for an opportunity to explore multiple return values, but...
In this case I'd just use a single "list of problems" return value, where naturally an empty list means no problem = success.
(Consistency across the whole API also matters, I haven't checked that)

Reply:
Yes, it might be simpler. Having that boolean value makes the return value more "semantic" to me. But I am not against taking a different approach. In any case, if we decide to go with just a list of errors, I would keep the JavaScript API as it is.

Reply:
Additionally, let's say that you could create the user but return a list of warnings (not errors at all). In that case, checking the first value is pretty straightforward (and you do not need to check the kind of errors you got).

Reply:
> Well, thanks for an opportunity to explore multiple return values, but... In this case I'd just use a single "list of problems" return value, where naturally an empty list means no problem = success. (Consistency across the whole API also matters, I haven't checked that)

Fine with me, in fact that is currently what is implemented in the assign_first_user backend method. @imobachgs what do you think?

Reply:
Sorry I read the cache version and did saw that you already replied :)

Reply:
> > Well, thanks for an opportunity to explore multiple return values, but... In this case I'd just use a single "list of problems" return value, where naturally an empty list means no problem = success. (Consistency across the whole API also matters, I haven't checked that)
> 
> Fine with me, in fact that is currently what is implemented in the assign_first_user backend method. @imobachgs what do you think?

I do not think it is needed, we do not need to expose exactly the same API that we have internally.